#include <iostream>
using namespace std;

void divide(int num, int den) {
    if (den == 0) {
        throw "Division by zero error";
    }
    cout << "Result of division: " << num / den << endl;
}

int main() {
    int numerator, denominator;

    cout << "Enter numerator: ";
    cin >> numerator;

    cout << "Enter denominator: ";
    cin >> denominator;

    try {
        divide(numerator, denominator);
    } catch (const char* error) {
        cout << "Exception caught: " << error << endl;
    }

    return 0;
}
